src/NHibernate/Cache/CacheLock.cs

```diff
@@ -85,10 +85,20 @@ public bool IsPuttable(long txTimestamp, object newVersion, IComparer comparator
 			{
 				return false;
 			}
-			return Version == null ?
```

Comment:
Seems `CachedItem` converted to lock `CacheLock` behaves differently for non-versioned entity.

`CachedItem` behaved as `minimalPut` is always true and `CacheLock` as `minimalPut` is always `false` (so `CacheLock` allows to update existing non-versioned entity, and CachedItem - always skips put for existing item)

Not sure if it's a bug - I kept this behavior for now in CacheLock.

P.S. Also I'm a bit confused how UnlockTimestamp and Timeout properties are used..

Reply:
> CacheLock behaves differently for non-versioned entity

Actually let's try to properly support it

Reply:
`UnlockTimestamp` initial value (-1) is irrelevant until Multiplicity has dropped to 0, at which point `UnlockTimestamp` is updated with the unlocking timestamp and starts to be relevant: it avoids a concurrent "stale" transaction to put in cache stale data.

The `Timeout` here is not actually a timeout, it is a `TimeoutTimestamp`: it has been computed from the locking timestamp to which the timeout has been added.

Reply:
> it avoids a concurrent "stale" transaction to put stale data in the cache.

So it's OK that `UnlockTimestamp`  is ignored for versioned entity?

Reply:
If the cached version is older than the one to be put yes. But with your change, it may be as old, in which case this is debattable.

Reply:
> But with your change, it may be as old, in which case this is debattable.

Why old? My change allows only to put the same version (`Compare` returns 0).

Reply:
I am just meaning the same age: cached version as old as the one to put, (or as new as,) that is just a way to tell "same age". This case is normally useful only for adding data for lazy loaded properties which was not already loaded and cached.

If the version is strictly newer, we know for sure that the cached data is stale and so, there is no harm in updating it with newer data, even if the session putting data in cache is not timestamped after the unlock timestamp. (Better to have less stale data.)

If the version is the same while the unlock timestamp tell us the session putting in cache may be stale, is it better to let other sessions see the additional date the put may add (for lazy properties), or let them hit the db?